app/src/store/stores/subServerStore.ts

```diff
@@ -0,0 +1,40 @@
+import { makeAutoObservable, runInAction } from 'mobx';
+import { Store } from 'store';
+import * as STATUS from 'types/generated/lit-status_pb';
+
+interface SubServers {
+  [key: string]: STATUS.SubServerStatus.AsObject;
+}
+
+export default class SubServerStore {
+  private _store: Store;
+
+  /** an array map of available subservers and their current status */
+  serverStatus: STATUS.SubServerStatusResp.AsObject = { subServersMap: [] };
+
+  constructor(store: Store) {
+    makeAutoObservable(this, {}, { deep: false, autoBind: true });
+
+    this._store = store;
+  }
+
+  /** getter to convert array map into an object keyed by subservers */
+  get subServers() {
+    const subServers: SubServers = {};
+
+    this.serverStatus.subServersMap.map(([serverName, serverStatus]) => {
+      subServers[serverName] = serverStatus;
+    });
+
+    return subServers;
+  }
+
+  /** fetch subserver statuses from the lit API */
+  async fetchStatus() {
+    const serverStatus = await this._store.api.lit.listSubServerStatus();
```

Comment:
Might make sense to have `try`/`catch` error handling here. Could also add a loading property to the class